concordium-contracts-common-derive/src/lib.rs

```diff
@@ -44,33 +44,50 @@ pub fn deserial_derive(input: TokenStream) -> TokenStream {
     unwrap_or_report(derive::impl_deserial(&ast))
 }
 
-/// Derive the Serial trait for the type.
+/// Derive the [`Serial`] trait for the type.
 ///
-/// If the type is a struct all fields must implement the Serial trait. If the
-/// type is an enum then all fields of each of the enums must implement the
-/// Serial trait.
+/// If the type is a struct all fields must implement the [`Serial`] trait. If
+/// the type is an enum then all fields of each of the enums must implement the
+/// [`Serial`] trait.
 ///
+/// Fields of structs are serialized in the order they appear in the code.
+///
+/// Enums can have no more than 65536 variants. They are serialized by using a
+/// tag to indicate the variant, enumerating them in the order they are written
+/// in source code. If the number of variants is less than or equal 256 then a
+/// single byte is used to encode it. Otherwise two bytes are used for the tag,
+/// encoded in little endian.
+///
+/// ## Generic type bounds
+///
+/// By default a trait bound is added on each generic type for implementing
+/// [`Serial`]. However, if this is not desirable, the bound can be put
+/// explicitly using the `bound` attribute on the type overriding the default
+/// behavior.
```

Comment:
Can you make it more explicit that this is added to the bounds present in the type declaration, i.e., if you have
```
#[concordium(T: Deserial)]
struct Foo<T: Deserial + Blah> {

}
```
then you get in total `T: Deserial + Blah + Deserial` as bounds (separated in where clause and type bounds).

I think a simple example is best really.

Reply:
In particular you cannot "get rid" of the type bounds using the `bound`.